cmd/okctl/forward_postgres.go

```diff
@@ -0,0 +1,241 @@
+		validation.Field(&o.Namespace, validation.Required),
+		validation.Field(&o.Username, validation.Required),
+		validation.Field(&o.PasswordFile, validation.Required),
+		validation.Field(&o.ListenPort, validation.Required),
+	)
+}
+
+// nolint: funlen
+func buildForwardPostgres(o *okctl.Okctl) *cobra.Command {
+	opts := &forwardPostgresOpts{}
+
+	cmd := &cobra.Command{
+		Use:   "postgres",
+		Short: "Forward to the given postgres database",
+		Args:  cobra.ExactArgs(0), // nolint: gomnd
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			if cmd.Name() == cobra.ShellCompRequestCmd {
+				return nil
+			}
+
+			var err error
+
+			err = loadUserData(o, cmd)
+			if err != nil {
+				return fmt.Errorf("loading application data: %w", err)
+			}
+
+			err = loadRepoData(o, cmd)
+			if err != nil {
+				if errors.Is(err, git.ErrRepositoryNotExists) {
+					return fmt.Errorf("okctl needs to be run inside a Git repository (okctl outputs " +
+						"various configuration files that will be stored here)")
+				}
+
+				return fmt.Errorf("loading repository data: %w", err)
+			}
+
+			o.Out = cmd.OutOrStdout()
+			o.Err = cmd.OutOrStderr()
+
+			return nil
+		},
+		PreRunE: func(_ *cobra.Command, _ []string) error {
+			err := o.InitialiseWithOnlyEnv(opts.Environment)
+			if err != nil {
+				return err
+			}
+
+			meta := o.RepoStateWithEnv.GetMetadata()
+			cluster := o.RepoStateWithEnv.GetCluster()
+			db := o.RepoStateWithEnv.GetDatabase(opts.ApplicationName)
+
+			opts.ID.Environment = opts.Environment
+			opts.ID.AWSAccountID = cluster.AWSAccountID
+			opts.ID.Repository = meta.Name
+			opts.ID.Region = meta.Region
+			opts.ID.ClusterName = o.RepoStateWithEnv.GetClusterName()
+			opts.Namespace = db.Namespace
+			opts.ConfigMapName = db.DatabaseConfigMapName
+			opts.SecretName = db.AdminSecretName
+			opts.SecurityGroup = db.SecurityGroupID
+
+			err = opts.Validate()
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
+		RunE: func(_ *cobra.Command, _ []string) (err error) {
+			clientSet, config, err := kube.NewFromEKSCluster(
+				opts.ID.ClusterName,
+				opts.ID.Region,
+				o.CloudProvider,
+				o.CredentialsProvider.Aws(),
+			).Get()
+			if err != nil {
+				return err
+			}
+
+			// Ensure that ENABLE_POD_ENI is true
+			err = awsnode.New(clientSet).EnablePodENI()
+			if err != nil {
+				return err
+			}
+
+			password, err := o.FileSystem.ReadFile(opts.PasswordFile)
+			if err != nil {
+				return err
+			}
+
+			app := fmt.Sprintf("%s-pgbouncer", opts.ApplicationName)
+
+			labels := map[string]string{
+				"pgbouncer": app,
+			}
+
+			policyClient := securitygrouppolicy.New(
+				app,
+				opts.Namespace,
+				securitygrouppolicy.Manifest(
+					app,
+					opts.Namespace,
+					labels,
+					[]string{opts.SecurityGroup},
+				),
+				config,
+			)
+
+			_, err = policyClient.Create()
+			if err != nil {
+				return err
+			}
+
+			defer func() {
```

Comment:
What happens if the user (sig)kills the terminal when in port-forwarding mode or the power switches off? Will it cause any problems the next time (s)he tries running `forward`?

If that's the case, we can _consider_ making `forward postgres` idempotent (possibly in a new PR).

Reply:
It won't be a problem, I have made it idempotent already, because I ran into the problem frequently when testing 🙈